### PR TITLE
Enginetest fails to link if OPENSSL_NO_ENGINE is defined

### PR DESCRIPTION
**Pull request by paulidale**
- [x] tests are added or updated

Also fixed an indent issue and a prototype one.

### CODE REVIEW

**Comment by paulidale** [comment]
Thanks.  Merged.

File: test/enginetest.c [hunk 1]
```diff
@@ -12,12 +12,13 @@
 #include <stdlib.h>
 #include <openssl/e_os2.h>
 
+# include "testutil.h"
```
Comment (kaduk) [inline test/enginetest.c]: I'm not sure it's worth another commit to remove the extra indentation in the copy/paste'd line.

Comment (paulidale) [inline test/enginetest.c]: I'll squash them.

Comment (paulidale) [inline test/enginetest.c]: oops, merged already.
I'll squash trivial ones like this in future.

**Pull request closed without merging**